src/WebJobs.Extensions.DurableTask/Bindings/OrchestrationTriggerAttributeBindingProvider.cs

```diff
@@ -87,7 +84,10 @@ private class OrchestrationTriggerBinding : ITriggerBinding
                 this.BindingDataContract = GetBindingDataContract(parameterInfo);
             }
 
-            public Type TriggerValueType => typeof(IDurableOrchestrationContext);
+            // Out-of-proc V2 uses a different trigger value type
+            public Type TriggerValueType => this.config.IsOutOfProcV2 ?
+                typeof(RemoteOrchestratorContext) :
+                typeof(IDurableOrchestrationContext);
```

Comment:
Does this have an effect on the expected contents of `function.json`? Namely, would it change the `binding->type` JSON field? Also, does this have implications for the development of the Java SDK?